File: admin/src/types/Blog.ts
export interface BlogRequest {
  title : string,
  author : string,
  slug : string,
  coverImagePath : string,
  tags : string,
  excerpt : string,
  content : string,
  publish : boolean,
  category : string
}

export interface BlogResponse {
  id : string,
  createdAt : string,
  title : string,
  author : string,
  slug : string,
  coverImagePath : string,
  coverImageUrl: string;
  tags : string,
  excerpt : string,
  content : string,
  category : string,
  publish : boolean
}


export interface BlogViewModel extends BlogResponse {
  tagsArray: string[];
  created: string;
}

export interface BlogUpdateRequest extends BlogRequest {
  id : string;
}

File: blogs_management/src/main/java/com/blogs_management/model/Blog.java
package com.blogs_management.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.OffsetDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "blogs")
public class Blog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY) // gen id ở phia DB : Insert xong → DB trả id về → Hibernate set lại entity
    private Long id;
    private String title;
    private String slug;
    private String author;
    private String category;
    @Column(name="cover_image_id")
    private Long coverImageId;
    private Boolean publish = true;

    @Column(columnDefinition = "TEXT")
    private String excerpt;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Column(name = "cover_image")
    private String coverImage;

    @Column(columnDefinition = "TEXT")
    private String tags;

    @Column(name = "created_at")
    @CreationTimestamp
    private OffsetDateTime createdAt;

    @Column(name = "updated_at")
    @UpdateTimestamp
    private OffsetDateTime updatedAt;
}
